sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/PartitionKey.java

```diff
@@ -30,6 +30,17 @@ public PartitionKey(final Object key) {
         this.internalPartitionKey = PartitionKeyInternal.fromObjectArray(new Object[] {key}, true);
     }
 
+    /**
+     * Constructor. CREATE a new instance of the PartitionKey object.
+     *
+     * @param keys the value of partition keys.
+     */
+    @SuppressWarnings("serial")
+    public PartitionKey(final Object[] keys) {
```

Comment:
Object can be any non primitive type, however for subpartitioning we only support primitive types,

we should consider using a builder pattern to construct and validate the content instead of passing raw object array.

Could you check what DotNet v3 is doing?

Reply:
Added the pattern as suggested.